test/e2e/network/loadbalancer.go

```diff
@@ -757,6 +822,13 @@ var _ = common.SIGDescribe("LoadBalancers", feature.LoadBalancer, func() {
 		e2epod.SetNodeSelection(&serverPod2.Spec, nodeSelection)
 		e2epod.NewPodClient(f).CreateSync(ctx, serverPod2)
 
+		// FIXME: this test is supposed to ensure that "a LoadBalancer UDP service
+		// doesn't blackhole the traffic to the node when the pod backend is
+		// destroyed and the traffic has to fall back to another pod", but it's
+		// possible that a connection to backend 2 will succeed at this point
+		// before backend 1 is deleted, in which case no further successful
+		// connections are required to pass.
```

Comment:
this can not happen, the test waits until pod1 is deleted and also validates the endpoints to avoid this problem

> 		e2epod.NewPodClient(f).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)

Reply:
It waits until pod 1 is deleted to start _checking_ if pod 2 has been reached, but it doesn't ensure that pod 2 was actually reached after pod 1 was deleted.

```
main thread                      DialUDP goroutine
-----------                      -----------------
"creating a backend pod"
"checking ... backend 1"
...                              DialUDP()
...                              hostnames.Insert("hostname1")
hostnames.Has("hostname1")
"creating a second backend pod"
                                 DialUDP()
                                 hostnames.Insert("hostname2")
DeleteSync
           (kube-proxy breaks here due to bugs)
                                 DialUDP()
                                 Logf("Failed to connect")
"checking ... backend 2"
hostnames.Has("hostname2")
Zarro boogs found!
```

Reply:
we use always the same source ip and source port (see L754) , you can't get to backend2 unless you use a round robin per packet for UDP loadbalancers, but well , that is not common and I would say this is safe assumption
 
```
 			laddr, err := net.ResolveUDPAddr("udp", ":54321")
			if err != nil {
				framework.Failf("Failed to resolve local address: %v", err)
			}
```